plugin/pkg/admission/noderestriction/admission.go

```diff
@@ -235,14 +236,21 @@ func (p *Plugin) admitPodStatus(nodeName string, a admission.Attributes) error {
 	switch a.GetOperation() {
 	case admission.Update:
 		// require an existing pod
-		pod, ok := a.GetOldObject().(*api.Pod)
+		oldPod, ok := a.GetOldObject().(*api.Pod)
 		if !ok {
 			return admission.NewForbidden(a, fmt.Errorf("unexpected type %T", a.GetOldObject()))
 		}
 		// only allow a node to update status of a pod bound to itself
-		if pod.Spec.NodeName != nodeName {
+		if oldPod.Spec.NodeName != nodeName {
 			return admission.NewForbidden(a, fmt.Errorf("node %q can only update pod status for pods with spec.nodeName set to itself", nodeName))
 		}
+		newPod, ok := a.GetObject().(*api.Pod)
+		if !ok {
+			return admission.NewForbidden(a, fmt.Errorf("unexpected type %T", a.GetObject()))
+		}
+		if !labels.Equals(oldPod.Labels, newPod.Labels) {
```

Comment:
It seems reflect.DeepEqual should be used here

Reply:
Why? `labels.Equals` is checking semantic equality and is much faster.

Reply:
I was looking at existing code.
e.g. in pkg/controller/daemon/daemon_controller.go , line 579:
```
	labelChanged := !reflect.DeepEqual(curPod.Labels, oldPod.Labels)
```

Reply:
Yeah, I think it's better to change the existing code to use `labels.Equals`, but it doesn't matter that much - both are logically equivalent.